scripts/cloudbuild_k8s_generator.py

```diff
@@ -1,261 +0,0 @@
-  args:
-  - -exc
-  - |
-    TAG="$$(cat ./MARKETPLACE_TOOLS_TAG)"
-    docker pull "gcr.io/cloud-marketplace-tools/k8s/dev:$$TAG"
-    docker tag "gcr.io/cloud-marketplace-tools/k8s/dev:$$TAG" "gcr.io/cloud-marketplace-tools/k8s/dev:local"
-
-- id: Get Kubernetes Credentials
-  name: gcr.io/cloud-builders/gcloud
-  waitFor:
-  - '-'
-  args:
-  - container
-  - clusters
-  - get-credentials
-  - '$_CLUSTER_NAME'
-  - --region
-  - '$_CLUSTER_LOCATION'
-  - --project
-  - '$PROJECT_ID'
-
-- id: Copy kubectl Credentials
-  name: gcr.io/google-appengine/debian9
-  waitFor:
-  - Get Kubernetes Credentials
-  entrypoint: bash
-  args:
-  - -exc
-  - |
-    mkdir -p /workspace/.kube/
-    cp -r $$HOME/.kube/ /workspace/
-
-- id: Copy gcloud Credentials
-  name: gcr.io/google-appengine/debian9
-  waitFor:
-  - Get Kubernetes Credentials
-  entrypoint: bash
-  args:
-  - -exc
-  - |
-    mkdir -p /workspace/.config/gcloud/
-    cp -r $$HOME/.config/gcloud/ /workspace/.config/
-
-- id: Install CRDs
-  name: gcr.io/cloud-marketplace-tools/k8s/dev:local
-  waitFor:
-  - Copy kubectl Credentials
-  - Copy gcloud Credentials
-  - Pull Dev Image
-  env:
-  - 'KUBE_CONFIG=/workspace/.kube'
-  - 'GCLOUD_CONFIG=/workspace/.config/gcloud'
-  # Use local Docker network named cloudbuild as described here:
-  # https://cloud.google.com/cloud-build/docs/overview#build_configuration_and_build_steps
-  - 'EXTRA_DOCKER_PARAMS=--net cloudbuild'
-  dir: k8s/{{ solution }}
-  args:
-  - make
-  - crd/install
-
-- id: Run diagnostic tool
-  name: gcr.io/cloud-marketplace-tools/k8s/dev:local
-  waitFor:
-  - Install CRDs
-  env:
-  - 'KUBE_CONFIG=/workspace/.kube'
-  - 'GCLOUD_CONFIG=/workspace/.config/gcloud'
-  # Use local Docker network named cloudbuild as described here:
-  # https://cloud.google.com/cloud-build/docs/overview#build_configuration_and_build_steps
-  - 'EXTRA_DOCKER_PARAMS=--net cloudbuild'
-  dir: k8s/{{ solution }}
-  args:
-  - make
-  - app/doctor
-
-- id: Build {{ solution }}
-  name: gcr.io/cloud-marketplace-tools/k8s/dev:local
-  waitFor:
-  - Run diagnostic tool
-  env:
-  - 'KUBE_CONFIG=/workspace/.kube'
-  - 'GCLOUD_CONFIG=/workspace/.config/gcloud'
-  # Use local Docker network named cloudbuild as described here:
-  # https://cloud.google.com/cloud-build/docs/overview#build_configuration_and_build_steps
-  - 'EXTRA_DOCKER_PARAMS=--net cloudbuild'
-  dir: k8s/{{ solution }}
-  args:
-  - make
-  - -j4
-  - app/build
-
-- id: Verify {{ solution }}
-  name: gcr.io/cloud-marketplace-tools/k8s/dev:local
-  waitFor:
-  - Build {{ solution }}
-  env:
-  - 'KUBE_CONFIG=/workspace/.kube'
-  - 'GCLOUD_CONFIG=/workspace/.config/gcloud'
-  # Use local Docker network named cloudbuild as described here:
-  # https://cloud.google.com/cloud-build/docs/overview#build_configuration_and_build_steps
-  - 'EXTRA_DOCKER_PARAMS=--net cloudbuild'
-  dir: k8s/{{ solution }}
-  args:
-  - make
-  - -j4
-  - app/verify
-
-{%- for extra_config in extra_configs %}
```

Comment:
Did you see this loop? Generally, we'd like to test some solution with different configuration, see L229.

We need this generate for it or we need to host Cloud Build configuration for each solution in the repo (like we do with triggers).

Reply:
There is no solution with that setup. Which solutions have that?

Reply:
Currently only WordPress, but nothing prevent us to add it to other solutions.

There was even a plan to extract the config to a file - https://github.com/GoogleCloudPlatform/click-to-deploy/issues/556